src/amber/router/router.cr

```diff
@@ -11,6 +11,19 @@ module Amber
         @routes = RouteSet(Route).new
         @routes_hash = {} of String => Route
         @socket_routes = Array(NamedTuple(path: String, handler: WebSockets::Server::Handler)).new
+        @scopes = [] of String
+      end
+
+      def push_scope(scope : String)
+        @scopes << scope
+      end
+
+      def pop_scope
+        @scopes.pop
+      end
+
+      private def actual_scope
```

Comment:
This probably doesn't need the `actual_` prefix, a private getter for just `scope` makes sense. The prefix `actual_` makes me think there'd be another method lying around called `fake_scope` or something.